app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,109 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all.order(:id)
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to task_not_found_path
+      return
+    end
+
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save # save returns true if the database insert succeeds
+      redirect_to task_path(@task) # go to the index so we can see the task in the list
+      return
+    else # save failed :(
+      render :new # show the new task form view again
+      return
+    end
+  end
+
+  def edit
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to task_not_found_path
+      return
+    else
+      render :new
+      return
+    end
```

Comment:
Rather than including code here to say `render  :new` it is convention to have a separate edit view. Since this view largely overlaps with the new view, this is a great place to use partial views.